sdkproject/Assets/Mapbox/Unity/Map/AbstractMap.cs

```diff
@@ -9,22 +9,43 @@
 	public abstract class AbstractMap : MonoBehaviour, IMap
 	{
 		[SerializeField]
+		[Range(0, 22)]
+		protected float _zoomRange;
+		public float ZoomRange
+		{
+			get
+			{
+				return _zoomRange;
+			}
+		}
+		public void SetZoomRange(float zoom)
+		{
+			_zoomRange = zoom;
+		}
+		[SerializeField]
 		bool _initializeOnStart = true;
 
 		[Geocode]
 		[SerializeField]
 		protected string _latitudeLongitudeString;
 
-		[SerializeField]
-		[Range(0, 22)]
-		protected int _zoom;
+
+		//private int _zoom;
 		public int Zoom
 		{
 			get
 			{
-				return _zoom;
+				return (int)Math.Ceiling(ZoomRange);
 			}
 		}
+        protected int _initialZoom;
```

Comment:
Please set your IDE to use tabs, not spaces, according to our code style guide:
https://github.com/mapbox/mapbox-unity-sdk/blob/develop/CODING-STYLE.md